src/pages/CreateOrderPage.jsx

```diff
@@ -115,23 +120,32 @@ const CreateOrderPage = () => {
     setShowAddAddressForm(false);
     dispatch(fetchAddresses());
   };
-  const handleSelectAddress = (address) => {
-    dispatch(setAddressAction(address));
+  const handleSelectAddress = (selectedId) => {
```

Comment:
Redux state güncellemeleri her zaman asenkron işlemlerdir. dispatch işlemi sonrası hemen yeni state değerlerine güvenmek yerine, bu değerlerin güncellenmesini beklemek için ilgili state değişikliklerini izleyen bir useEffect kullanabilirsin.